recipes-extended/qemu-dm/qemu-dm/qmp-argo-char-driver.patch

```diff
@@ -85,20 +85,37 @@
 +static int argo_chr_write(Chardev *chr, const uint8_t *buf, int len)
 +{
 +    ArgoChardev *s = ARGO_CHARDEV(chr);
-+    int ret;
++    int write_len = len;
++    int written = 0;
 +
-+    if (s->stream) {
-+        ret = argo_send(s->fd, buf, len, 0);
-+    } else {
-+        ret = argo_sendto(s->fd, buf, len, 0, &s->remote_addr);
-+    }
-+    if (ret != len) {
-+        fprintf(stderr, "%s error: argo_sendto() failed (%s) - %d %d.\n",
-+                ARGO_CHARDRV_NAME, strerror(errno), ret, len);
-+        return 0;
++    while ( written < len ) {
++        int ret;
++
++        if (s->stream) {
++            ret = argo_send(s->fd, buf, write_len, 0);
++        } else {
++            ret = argo_sendto(s->fd, buf, write_len, 0, &s->remote_addr);
++        }
++
++        if ( ret == -1 && errno == EMSGSIZE) {
++            /* If the write was too big, try a smaller piece */
++            write_len /= 2;
++            continue;
++        } else if (ret == -1) {
++            fprintf(stderr, "%s error: argo_sendto() failed (%s) - %d %d.\n",
++                    ARGO_CHARDRV_NAME, strerror(errno), ret, len);
```

Comment:
This PR has really good improvements to the code - I especially like the use of clear variable names and the logic to step down the size of the workload to the point that the receiver has available capacity for. Thanks for doing this.
The only (super minor) point I can see in this section is about the error message in the above line -- which was inherited from the prior code: it's not always the case that `argo_sendto` is the function that was invoked -- ie. `argo_send` is called instead for streams. Is the existing error message close enough? Up to you - the improvements here are worthwhile anyway.